appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go

```diff
@@ -386,6 +395,18 @@ func generateDesiredResource(ctx context.Context, env appstudioshared.Environmen
 			Namespace: env.Namespace,
 		},
 	}
+
+	managedEnvSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateManagedEnvSecretName(env.Name),
+			Namespace: secret.Namespace,
+			Labels: map[string]string{
+				managedEnvironmentSecretLabel: env.Name,
+			},
```

Comment:
Thinking out loud, could we add an ownerref from the  Secret back to the Environment? This might aid cleanup, and I don't think there are any cases where the lifecycle of the ManagedEnvironment Secret needs to outlive the Environment itself.